dll/win32/shell32/folders/CFSFolder.cpp

```diff
@@ -136,10 +136,13 @@ HRESULT GetCLSIDForFileType(PCUIDLIST_RELATIVE pidl, LPCWSTR KeyName, CLSID* pcl
 static HRESULT
 getDefaultIconLocation(LPWSTR szIconFile, UINT cchMax, int *piIndex, UINT uFlags)
 {
-    if (!HCR_GetIconW(L"Folder", szIconFile, NULL, cchMax, piIndex))
+    if (!HLM_GetIconW(IDI_SHELL_FOLDER - 1, szIconFile, cchMax, piIndex))
     {
-        lstrcpynW(szIconFile, swShell32Name, cchMax);
-        *piIndex = -IDI_SHELL_FOLDER;
+        if (!HCR_GetIconW(L"Folder", szIconFile, NULL, cchMax, piIndex))
+        {
+            lstrcpynW(szIconFile, swShell32Name, cchMax);
```

Comment:
I know you just kept the original code, but otherwise,
```suggestion
            StringCchCopyW(szIconFile, cchMax, swShell32Name);
```